ADALiOS/ADALiOS/ADKeychainTokenCacheStore.m

```diff
@@ -258,12 +269,14 @@ - (ADTokenCacheStoreItem*)itemFromKeyhainAttributes:(NSDictionary*)attrs
     for (NSDictionary* attrs in items)
     {
         ADTokenCacheStoreItem* item = [self itemFromKeyhainAttributes:attrs];
-        if (!item)
+        if (item)
         {
-            continue;
+            BOOL validity =  fromGraveyard ? [item markAsDead] : ![item markAsDead];
```

Comment:
Let higher layers of code decide whether or not they want to deal with tombstoned items.